docs/surefire-providers.adoc

```diff
@@ -0,0 +1,45 @@
+== Surefire Providers
+
+Internally, Smart Testing uses its own implementation of `SurefireProvider` where calls Smart Testing API. Then, based on the information retrieved from the classpath, it decides which provider should be used for a delegation of the test execution - eg. JUnit4 or TestNG (in the same way as it is done inside of the Surefire plugin itself).
+But some projects may use their own surefire provider implementations or can specify one specific known provider that should scan the classpath and execute the test suite.
+
+=== Known Surefire Providers
+
+Known Surefire Providers are those implementations that are part of the Surefire project itself plus one implementation distributed as part of JUnit 5. The complete list of known providers:
+
+* JUnit 4
+** groupId: `org.apache.maven.surefire`
+** artifactId: `surefire-junit4`
+** provider implementation: `org.apache.maven.surefire.junit4.JUnit4Provider`
+
+* JUnit 47 and above
+** groupId: `org.apache.maven.surefire`
+** artifactId: `surefire-junit47`
+** provider implementation: `org.apache.maven.surefire.junitcore.JUnitCoreProvider`
+
+* JUnit 5
+** groupId: `org.junit.platform`
+** artifactId: `junit-platform-surefire-provider`
+** provider implementation: `org.junit.platform.surefire.provider.JUnitPlatformProvider`
+
+* TestNG
+** groupId: `org.apache.maven.surefire`
+** artifactId: `surefire-testng`
+** provider implementation: `org.apache.maven.surefire.testng.TestNGProvider`
+
+When one of these providers is specified inside of the Surefire's dependencies, then Smart Testing is able to automatically detect it and delegate the execution to it.
+
+=== Custom Surefire Providers
+
+If some unknown provider (or some other `SurefireProvider` implementation) is used then it is necessary to set this provider either in the configuration file or using system properties.
+The format is a pair of groupId and artifactId with a fully qualified name of the class that implements `SurefireProvider` interface and that should be loaded. For JUnit 5 provider it would look like this (in case of configuration file):
+```yaml
+customProviders:
+    - org.junit.platform:junit-platform-surefire-provider=org.junit.platform.surefire.provider.JUnitPlatformProvider
+```
+If this is specified then Smart Testing will detect the dependency and delegate the test execution to it.
+System property dedicated for this usage is `const:core/src/main/java/org/arquillian/smart/testing/configuration/Configuration.java[name="SMART_TESTING_CUSTOM_PROVIDERS"]`
+
+For now, only one custom provider is supported. In next releases, we will bring you a support for more custom providers.
```

Comment:
Not sure if this is helpful information. Maybe mention which one is supported?

Reply:
Yeah, I wrote it probably in a bad way.
I wanted to say that the delegation is supported only for one custom provider (no matter which one). This means that if you specify more custom providers in the configuration (using `customProviders`), then only one will be used. https://github.com/arquillian/smart-testing/issues/282

Reply:
Yeah, I'll remove it. The probability of multiple custom providers in one surefire config is probably equal to zero.